sync-core/src/main/java/com/cloudant/sync/datastore/SavedHttpAttachment.java

```diff
@@ -49,23 +48,26 @@
      */
     public SavedHttpAttachment(String name, Map<String,Object> attachmentData, URI attachmentURI)
            throws IOException {
-         super(name, (String)attachmentData.get("content_type"), Encoding.Plain);
-         Boolean stub = (Boolean) attachmentData.get("stub");
-         Number length = (Number)attachmentData.get("length");
+         super(name,
+                 (String)attachmentData.get("content_type"),
+                 Attachment.getEncodingFromString((String)attachmentData.get("encoding")));
+
+         Number length;
          String data = (String)attachmentData.get("data");
-         String encoding =  (String)attachmentData.get("encoding");
-         this.encoding = Attachment.getEncodingFromString(encoding);
-         if(!stub){
+         if(data != null){
             byte[] dataArray =  data.getBytes();
             InputStream is = Base64InputStreamFactory.get(new ByteArrayInputStream(dataArray));
             this.data = IOUtils.toByteArray(is);
+            length = this.data.length;
+         } else {
+             length = (Number)attachmentData.get("length");
          }
 
          this.attachmentURI = attachmentURI;
          this.size = length.intValue();
```

Comment:
Is it possible to declare `size` as `long` and use `longValue()`? We store sizes as `long`s in most places although I appreciate users with >4GB attachments are probably asking for trouble.